### PR TITLE
Fix PropertyAccess crash in comment reference

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: lib/src/comment_references/model_comment_reference.dart [hunk 1]
```diff
@@ -123,3 +123,18 @@ class _ModelCommentReferenceImpl implements ModelCommentReference {
   late final List<CommentReferenceNode> parsed =
       CommentReferenceParser(codeRef).parse();
 }
+
+extension on CommentReferableExpression {
+  Element? get element {
+    var expression = this;
+    if (expression is PropertyAccess) {
```
Comment (scheglov) [inline lib/src/comment_references/model_comment_reference.dart:130]: Might be better to sort these `if is X`.

File: lib/src/comment_references/model_comment_reference.dart [hunk 2]
```diff
@@ -123,3 +123,18 @@ class _ModelCommentReferenceImpl implements ModelCommentReference {
   late final List<CommentReferenceNode> parsed =
       CommentReferenceParser(codeRef).parse();
 }
+
+extension on CommentReferableExpression {
+  Element? get element {
+    var expression = this;
```
Comment (scheglov) [inline lib/src/comment_references/model_comment_reference.dart:129]: This name is fine, just I just note that in the analyzer we tend to name it `final self = this;`.

**Review by scheglov — APPROVED** [review]

File: lib/src/comment_references/model_comment_reference.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/comment_references/model_comment_reference.dart]: done

File: lib/src/comment_references/model_comment_reference.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/comment_references/model_comment_reference.dart]: self is better; some of these sub-types have an expression field.

**Pull request merged**